src/web_fetch.js

```diff
@@ -0,0 +1,159 @@
+const consts = require('./consts');
+const errConst = require('./error_const');
+const makeError = require('./native/_error.js');
+const { parse: parseUrl } = require('url');
+const mime = require('mime');
+const nodePath = require('path');
+
+/**
+* @typedef {Object} WebFetchOptions
+* holds additional options for the `webFetch` function.
+* @param {Object} range range of bytes to be retrieved.
+* The `start` attribute is expected to be the start offset, while the
+* `end` attribute of the `range` object the end position (both inclusive)
+* to be retrieved, e.g. with `range: { start: 2, end: 3 }` the 3rd
+* and 4th bytes of data will be retrieved.
+* If `end` is not specified, the bytes retrived will be from the `start` offset
+* untill the end of the file.
+* The ranges values are also used to populate the `Content-Range` and
+* `Content-Length` headers in the response.
+*/
+
+/**
+* Helper to lookup a given `safe://`-url in accordance with the
+* convention and find the requested object.
+*
+* @param {String} url the url you want to fetch
+* @param {WebFetchOptions} [options=null] additional options
+* @returns {Promise<Object>} the object with body of content and headers
+*/
+async function webFetch(url, options) {
```

Comment:
Let's keep a consistency with using arrow functions.

Reply:
Oh I forgot. The reason this is specifically a function declaration is because it is must be bound to the SAFEApp instance, otherwise arrow functions cannot bind `this`.